plugins/module_utils/redfish_utils.py

```diff
@@ -767,7 +767,7 @@ def get_disk_inventory(self, systems_uri):
         properties = ['BlockSizeBytes', 'CapableSpeedGbs', 'CapacityBytes',
                       'EncryptionAbility', 'EncryptionStatus',
                       'FailurePredicted', 'HotspareType', 'Id', 'Identifiers',
-                      'Manufacturer', 'MediaType', 'Model', 'Name',
+                      'Links', 'Manufacturer', 'MediaType', 'Model', 'Name',
```

Comment:
What's driving the need to add "Links" to this? Links will just contain a set of URIs to different areas of the model. How would a user consume this data if they aren't actively interacting with the service?

Reply:
Actually, to create volumes [6813](https://github.com/ansible-collections/community.general/pull/6813) we need an input which looks like 
`Name: "MR Volume"
          RAIDType: "RAID0"
          Links:
            Drives:
              - "@odata.id": "/redfish/v1/Systems/1/Storage/DE00B000/Drives/1"
`
To populate the drives for this, it would be convenient to get the link from the GetDiskInventory operation, as show below.
`
"Links": {
                    "Storage": {
                        "@odata.id": "/redfish/v1/Systems/1/Storage/DE00B000"
                    },
                    "Volumes": [
                        {
                            "@odata.id": "/redfish/v1/Systems/1/Storage/DE00B000/Volumes/3"
                        }
                    ],
                    "Volumes@odata.count": 1
                }
`

Reply:
any updates on this?

Reply:
I would recommend we flatten this a bit to focus on what's useful to the user. We don't need to copy the Redfish model as-is.

I would suggest bringing up the links the user cares about to the root of the object. So, the results object would look like this:

```
BlockSizeBytes: 512
FailurePredicted: false
PartNumber: 123456789ABCDEF
SerialNumber: 123456789ABCDEF
Volumes:
  - "@odata.id": "/redfish/v1/Systems/1/Storage/DE00B000/Volumes/1"
```

I would also give the same recommendation for "CreateVolume".

Reply:
Agreed, Changed as suggested. This has also been changed for "CreateVolume".

Reply:
Resolving this conversation. Thanks!